docs/writing-tests/testharness-api.md

```diff
@@ -430,18 +430,39 @@ not possible to make the test reliable in some other way.
 
 ## Setup ##
 
-Sometimes tests require non-trivial setup that may fail. For this purpose
-there is a `setup()` function, that may be called with one or two arguments.
-The two argument version is:
+Sometimes tests require non-trivial setup that may fail. testharness.js
+provides two global functions for this purpose, `setup` and `promise_setup`.
+
+`setup()` may be called with one or two arguments. The two argument version is:
 
 ```js
 setup(func, properties)
 ```
 
-The one argument versions may omit either argument.
-func is a function to be run synchronously. `setup()` becomes a no-op once
-any tests have returned results. Properties are global properties of the test
-harness. Currently recognised properties are:
+The one argument versions may omit either argument. `func` is a function to be
+run synchronously. `setup()` becomes a no-op once any tests have returned
+results. `properties` is an object which specifies global properties of the
+test harness (enumerated in the following section).
+
+`promise_setup()` allows authors to pause testing until some asynchronous
+operation has completed. It has the following signature:
+
+```js
+promise_setup(func, properties)
+```
+
+Here, the `func` argument is required. The harness will defer execution of this
+function until any subtests have completed. The function must return an object
```

Comment:
Its any *previous* (ongoing?) subtests, right? Also, what about previous `async_test` subtests?

Reply:
Actually, performing "setup" after testing has begun contradicts the notion of
"setup." Above, @zcorpan expressed an intent to formally disallow it, so I
think we should avoid documenting (and implicitly endorsing) the pattern.

Reply:
That SGTM.